addons/account/models/account_move.py

```diff
@@ -683,6 +687,13 @@ def _compute_date(self):
                 # might be protected because `_get_accounting_date` requires the `name`
                 self.env.add_to_compute(self._fields['name'], move)
 
+    @api.depends('move_type', 'journal_id')
+    def _compute_invoice_cash_rounding_id(self):
+        for record in self:
+            if record.invoice_cash_rounding_id:
+                continue
```

Comment:
So, taking the example given by @fpodoo for Belgium
* Select Cash Journal (applies rounding)
* Select Bank Journal (no rounding necessary)

We end up with the rounding for the cash journal